arbor/simulation.cpp

```diff
@@ -253,6 +263,9 @@ void simulation_state::update(const connectivity& rec) {
     // Forget old generators, if present
     event_generators_.clear();
     event_generators_.resize(num_local_cells);
+    scratch_.clear();
+    scratch_.resize(num_local_cells);
```

Comment:
this will deallocate - intended? Why not `resize` followed by `clear` over the elements?

Reply:
Not necessarily, it's not guaranteed by the standard ;) I'll check how I can make that better.